### PR TITLE
Reader typings

### PR DESCRIPTION
**Pull request by WojciechP**

### CODE REVIEW

File: src/main/javascript/monet.d.ts [hunk 1]
```diff
@@ -298,14 +298,26 @@ declare namespace monet {
...
+        chain<B>(fn: (val: A) => Reader<E, B>): Reader<E, B>;
+        map<B>(fn: (val: A) => B): Reader<E, B>;
+        join<B>(): Reader<E, B>; // if A is Reader<B>
+        takeLeft<X>(m: Reader<E, X>): Reader<E, A>;
+        takeRight<B>(m: Reader<E, B>): Reader<E, B>;
+        ap<B>(rfn: Reader<E, (val: A) => B>): Reader<E, B>;
+
+        /* Reader-specific: */
         run(config: E): A;
+        local<X>(fn: (val: X) => E): Reader<X, A>;
```
Comment (ulfryk) [inline src/main/javascript/monet.d.ts:314]: what about:
- init
- of
- point
- pure
- unit

```
```

File: tsconfig.json [hunk 2]
```diff
@@ -2,6 +2,7 @@
   "files": [
     "src/main/javascript/monet.d.ts",
     "src/test/typescript/identity-spec.ts",
-    "src/test/typescript/monad-laws.ts"
+    "src/test/typescript/monad-laws.ts",
```
Comment (ulfryk) [inline tsconfig.json]: add Reader to monad-laws tests also ;p

File: src/main/javascript/monet.d.ts [hunk 3]
```diff
@@ -298,14 +298,26 @@ declare namespace monet {
...
     }
     
     interface ReaderStatic {
         <E, A>(fn: (E) => A): Reader<E, A>;
+        of<E, A>(val: A): Reader<E, A>;
+        ask<E>(): Reader<E, E>;
```
Comment (ulfryk) [inline src/main/javascript/monet.d.ts:323]: pleas add the `new` based constructor also

File: src/main/javascript/monet.d.ts (hunk 1 above)
Comment (WojciechP) [inline src/main/javascript/monet.d.ts:314]: `init` is plain harmful, so I'm against exposing it.
The other ones are constructors - they are typed on `ReaderStatic`.

Comment (ulfryk) [inline src/main/javascript/monet.d.ts:314]: :+1:

**Pull request merged**